src/ripple/basics/base_uint.h

```diff
@@ -177,22 +178,82 @@ class base_uint
         memcpy(data_.data(), data, bytes);
     }
 
+    // Helper function to initialize a base_uint from a std::string_view.
+    constexpr decltype(data_)
+    initFromStringView(std::string_view sv)
+    {
+        // Local lambda that converts a single hex char to four bits.
+        // Return is a uint32_t to avoid integer conversions.
+        auto hexCharToUInt = [](char c) -> std::uint32_t {
+            std::uint32_t ret = 0xFFu;
+            if (c < '0' || c > 'f')
+                Throw<std::range_error>("invalid hex character");
+
+            if (c >= 'a')
+                ret = static_cast<std::uint32_t>(c - 'a' + 0xA);
+            else if (c >= 'A')
+                ret = static_cast<std::uint32_t>(c - 'A' + 0xA);
+            else if (c <= '9')
+                ret = static_cast<std::uint32_t>(c - '0');
+
+            if (ret > 0xFu)
+                Throw<std::range_error>("invalid hex character");
+
+            return ret;
+        };
+
+        decltype(data_) ret{};
+
+        if (sv == "0")
+        {
+            return ret;
+        }
+
+        if (sv.size() != size() * 2)
+            Throw<std::invalid_argument>("invalid length for hex string");
+
+        auto out = ret.data();
+
+        auto in = sv.begin();
+
+        while (in != sv.end())
+        {
+            std::uint32_t accum = {};
+            accum |= hexCharToUInt(*in++) << 4;
+            accum |= hexCharToUInt(*in++) << 0;
+            accum |= hexCharToUInt(*in++) << 12;
+            accum |= hexCharToUInt(*in++) << 8;
+            accum |= hexCharToUInt(*in++) << 20;
+            accum |= hexCharToUInt(*in++) << 16;
+            accum |= hexCharToUInt(*in++) << 28;
+            accum |= hexCharToUInt(*in++) << 24;
+            *out++ = accum;
+        }
+        return ret;
+    }
+
 public:
-    base_uint()
+    constexpr base_uint() : data_{}
     {
-        *this = beast::zero;
     }
 
-    base_uint(beast::Zero)
+    constexpr base_uint(beast::Zero) : data_{}
     {
-        *this = beast::zero;
     }
 
     explicit base_uint(std::uint64_t b)
```

Comment:
This is probably easy to also turn into a `constexpr`, but I'm not sure if I'd prefer to do that or eliminate this constructor altogether.

Reply:
This constructor is not easy to turn `constexpr` since the implementation currently calls `boost::endian::native_to_big()` which is not `constexpr` until boost 1.72.0 for gcc and clang.  It also appears that `native_to_big()` is not `constexpr` for MSVC in any boost release.